src/content/reference/react/useEffect.md

```diff
@@ -66,7 +66,7 @@ function ChatRoom({ roomId }) {
 
 * If your Effect is caused by an interaction (like a click), **React may run your Effect before the browser paints the updated screen**. This ensures that the result of the Effect can be observed by the event system. Usually, this works as expected. However, if you must defer the work until after paint, such as an `alert()`, you can use `setTimeout`. See [reactwg/react-18/128](https://github.com/reactwg/react-18/discussions/128) for more information.
 
-* Even if your Effect was caused by an interaction (like a click), **React may allow the browser to repaint the screen before processing the state updates inside your Effect.** Usually, this works as expected. However, if you must block the browser from repainting the screen, you need to replace `useEffect` with [`useLayoutEffect`.](/reference/react/useLayoutEffect)
+* Even if your Effect was caused by an interaction (like a click), **React may allow the browser to repaint the screen before executing the code inside your Effect that responds to state updates.** Usually, this works as expected. However, if you must block the browser from repainting the screen, you need to replace `useEffect` with [`useLayoutEffect`.](/reference/react/useLayoutEffect)
```

Comment:
Explained on the issue, but this isn't correct. The point here is that even if effect fire before paint, any state updates inside the effect are rendered after paint.